packages/credentials/src/V1/KiltCredentialV1.ts

```diff
@@ -354,6 +416,7 @@ export async function validateSubject(
   if (!credentialsCTypeId) {
     throw new Error('credential type does not contain a valid CType id')
   }
+
   // check that we have access to the right schema
   let cType = cTypes?.find(({ $id }) => $id === credentialsCTypeId)
   if (!cType) {
```

Comment:
We need to somehow handle both the ctypes passed in as the `cTypes` argument, and the `loadCtypes` function, if there is any. I suggest we create a combined ctype loader function from both, which first checks if the cType whose id is being looked up is contained in the cTypes array, and if not, forwards the id to the `loadCtypes` function, which then tries to look it up.

Hint: checking if we have a CType already in `cTypes` can be done quicker if we transform the array into a Map of ctype id to ctype definition once.

Reply:
I am working on a new commit for this.

Reply:
this is still tbd